apps/golden-sample-app/src/app/interceptors/ach-positive-pay.interceptor.ts

```diff
@@ -30,6 +33,13 @@ export class AchPositivePayInterceptor implements HttpInterceptor {
       );
     }
 
-    return next.handle(req);
+    return next.handle(req).pipe(
+      catchError((err: HttpErrorResponse) => {
+        if (err.status.toString().startsWith('4')) {
+          this.router.navigateByUrl('/error-page');
```

Comment:
What are we demonstrating here?

Reply:
There was a use case for handling HTTP errors with routing .

Reply:
What will happen if for any reason we get 5xx errors?

Reply:
Maybe also we can pass a custom message in extra params to demonstrate how to use `state` option: https://angular.io/api/router/NavigationBehaviorOptions#state and also we can try `skipLocationChange` https://angular.io/api/router/NavigationBehaviorOptions#skipLocationChange

Reply:
As discussed, I have removed this. First, We should understand how Journey's implement error handling. We can have a separate task for this.